pgcli/main.py

```diff
@@ -433,7 +433,10 @@ def connect(self, database='', host='', user='', port='', passwd='',
                 pgexecute = PGExecute(database, user, passwd, host, port, dsn,
                                       application_name='pgcli', **kwargs)
                 if passwd:
-                    keyring.set_password('pgcli', key, passwd)
+                    try:
+                        keyring.set_password('pgcli', key, passwd)
+                    except keyring.errors.InitError:
```

Comment:
What happens when we silence this `InitError`? It looks like we don't fall back to anything else, the connection just silently fails?

Reply:
No, if I understand #873 correctly it stores the password as a convenience. The down side of this fix is that the user doesn't know why his password isn't stored. On the other side, if the user didn't read the source of pgcli he probably wouldn't even know his password is being saved.

I think this feature (saving passwords) need some work, this is a quick workaround for a failing pgcli on my system. (I should have mentioned this first, sorry)

Reply:
Got it. It may not be perfect, better than it was before, for sure. Let's merge it.